### PR TITLE
Feature/89 nshmmodel init

### LINKED ISSUE #89: classmethod for `NshmModel` to build model from logic trees and config rather than files
Issue 1: User may want to construct a `NshmModel` object on the fly rather than defining files or using `get_model_version()`. 
Issue 2: if a user gets a model and then tries to modify the logic trees this results:

```
from nzshm_model import get_model_version
model = get_model_version('NSHM_v1.0.4')
slt = model.source_logic_tree
slt.branch_sets = [bs for bs in model.source_logic_tree.branch_sets if bs.short_name=="CRU"]
model.source_logic_tree.branch_sets = [bs for bs in model.source_logic_tree.branch_sets if bs.short_name=="CRU"]
print(len(slt.branch_sets))
print(len(model.source_logic_tree.branch_sets))
```

```
$ poetry run python demo/adapters_demo.py 
1
4
```

The logic tree attribues of an `NshmModel` object cannot be modified because they are loaded from files every time.

Solution: 
- [ ] init `NshmModel` from objects:
```
def __init__(self, slt:  gmcmlt: config: Optional[Config], title, etc)
    ...
```

- [ ] classmethod constructor to biuld from files, as is currently done
```
@classmethod
def from_files(cls,  self, version: str, title: str,  slt_json: Union[str, Path],  gmm_json: Union[str, Path],  gmm_xml: Union[str, Path],  hazard_config: 'HazardConfig'):
    ...
```

- [ ] update tests so that we're not jumping though hoops to create a NshmModel instance

### PR DESCRIPTION
**Pull request by chrisdicaprio**
closes #89 and #95

### CODE REVIEW

File: nzshm_model/model.py [hunk 1]
```diff
@@ -33,92 +28,69 @@ def __init__(
...
         Returns:
-            a gmm_logic_tree.
+            a source logic tree
+
         """
-        doc = NrmlDocument.from_xml_file(self._gmm_xml)
-        return doc.logic_trees[0]
+        return SourceLogicTree.from_source_logic_tree(SourceLogicTreeV1.from_json(filepath))
 
     @classmethod
```
Comment (chrisbc) [inline nzshm_model/model.py:95]: yetanother @staticmethod

**Review by chrisbc — APPROVED** [review]
cool @chrisdicaprio  - just the static vs classmethod issues noted

File: nzshm_model/logic_tree/logic_tree_base.py [hunk 2]
```diff
@@ -139,7 +140,7 @@ def composite_branches(self) -> Generator[CompositeBranch, None, None]:
             yield composite_branch
 
     @classmethod
```
[the remark this answers is not in the document]
Comment (chrisdicaprio) [reply nzshm_model/logic_tree/logic_tree_base.py:142]: this needs to be `@classmethod` to allow child classes to use the method

File: nzshm_model/model.py [hunk 3]
```diff
@@ -33,92 +28,69 @@ def __init__(
...
-
-    @property
-    def _glt_data(self):
-        with open(self._gmm_json, 'r') as jsonfile:
-            data = json.load(jsonfile)
-        return data
-
-    @property
-    def source_logic_tree(self) -> SourceLogicTree:
+    @classmethod
```
[the remark this answers is not in the document]
Comment (chrisdicaprio) [reply nzshm_model/model.py:49]: agreed to continue to use `@classmethod` to preserve pattern and make method's application more obvious

File: nzshm_model/model.py (hunk 1 above)
Comment (chrisdicaprio) [reply nzshm_model/model.py:95]: agreed to continue to use `@classmethod` to preserve pattern and make method's application more obvious

**Pull request merged**